lib/send-http-request.js

```diff
@@ -111,6 +111,20 @@ module.exports = {
     // Ensure method is upper-cased.
     inputs.method = inputs.method.toUpperCase();
 
+    // Check for a `content-type` (case-insensitive) header
+    if(!_.contains(['GET','HEAD','OPTIONS'], inputs.method) && inputs.headers) {
+      for (let headerName in inputs.headers) {
+        // If we found one, and it's one of our
+        if(headerName.toLowerCase() === 'content-type') {
+          let SUPPORTED_ENCTYPES = ['application/json', 'application/x-www-form-urlencoded', 'multipart/form-data'];
+          let contentType = inputs.headers[headerName];
+          if(_.contains(SUPPORTED_ENCTYPES, contentType) && (inputs.enctype !== contentType)) {
+            throw new Error(`Could not determine how to encode data in sendHttpRequest(), because the 'content-type' header does not match the provided \`enctype\` ('${inputs.enctype}').\nTo resolve this, you can do one of the following:\n• Use \`enctype: '${contentType}'\`\n• Set the 'content-type' header to '${inputs.enctype}'`);
```

Comment:
but if enctype is undefined, and content-type is set to 'application/x-www-form-urlencoded' or 'multipart/form-data'...